src/KubernetesClient/Authentication/TokenFileAuth.cs

```diff
@@ -0,0 +1,31 @@
+using System;
+using System.Net.Http.Headers;
+using System.IO;
+using System.Threading;
+using System.Threading.Tasks;
+using Microsoft.Rest;
+
+namespace k8s.Authentication
+{
+    public class TokenFileAuth : ITokenProvider
+    {
+        private string _token;
+        internal string _token_file { get; set; }
+        internal DateTime _token_expires_at { get; set; }
+
+        public TokenFileAuth(string token_file)
+        {
+            _token_file = token_file;
+        }
+
+        public async Task<AuthenticationHeaderValue> GetAuthenticationHeaderAsync(CancellationToken cancellationToken)
+        {
+            if (_token_expires_at < DateTime.UtcNow)
+            {
+                _token = File.ReadAllText(_token_file).Trim();
+                _token_expires_at = DateTime.UtcNow.AddSeconds(60);
```

Comment:
Please add a comment about why 60 seconds is the right value.

Reply:
done